test/unit/serialization/compact/CompactGenericRecordTest.js

```diff
@@ -568,6 +589,62 @@ describe('CompactGenericRecordTest', function () {
             });
         });
 
+        describe('array restrictions', function () {
+            it('should not throw error if object types are equal on ARRAY_OF_COMPACT', function () {
+                (async () => {
+                    const {serializationService, schemaService} = createSerializationService(
+                        [new ArrayOfCompactSerializer(), new SampleObject1Serializer()]
+                    );
+                    const object1 = new SampleObject1('type1', Long.fromNumber(102310312));
+                    const object2 = new SampleObject1('type2', Long.fromNumber(102310312));
+                    const arrayOfObjects = [
+                        object1,
+                        object2
+                    ];
+                    const arrayOfCompactObject = new ArrayOfCompact(arrayOfObjects);
+                    await serialize(serializationService, schemaService, arrayOfCompactObject);
+                }).should.not.throw();
+            });
+            it('should throw error if object types are not equal on ARRAY_OF_COMPACT', async function () {
+                const {serializationService, schemaService} = createSerializationService(
+                    [new ArrayOfCompactSerializer(), new SampleObject1Serializer(), new SampleObject2Serializer()]
+                );
+                const object1 = new SampleObject1('type1', Long.fromNumber(102310312));
+                const object2 = new SampleObject2('name1', Long.fromNumber(102310312));
+                const arrayOfObjects = [
+                    object1,
+                    object2
+                ];
+                const arrayOfCompactObject = new ArrayOfCompact(arrayOfObjects);
+
+                const error = await TestUtil.getRejectionReasonOrThrow(async () => {
+                    await serialize(serializationService, schemaService, arrayOfCompactObject);
+                });
+
+                error.should.be.instanceOf(HazelcastSerializationError);
+                error.message.includes('It is not allowed to serialize an array of Compact serializable objects'
+                +' containing different item types.').should.be.true;
+            });
+            it('should not throw error array of GenericRecord objects containing same schemas.', function () {
+                (async () => {
+                    const {serializationService, schemaService} = createSerializationService();
+                    const arrayofGenericRecords = getGenericRecordArray('same');
+                    await serialize(serializationService, schemaService, arrayofGenericRecords);
+                }).should.not.throw();
```

Comment:
If the test throws it will fail anyway. Therefore I don't think `(async () => {}).should.not.throw()` is necessary here.

Reply:
I removed unnecessary parts.